src/DataBase/EntryDatabase.cs

```diff
@@ -142,49 +142,30 @@ public async Task<bool> DeleteAsync(Expression<Func<IEntryModel, bool>> filter,
 
         public async Task<bool> DeleteListAsync(Expression<Func<IEntryModel, bool>> filter, EntryType entryType)
         {
-            if (entryType == EntryType.Income)
+            IEnumerable<IEntryModel> entriesToRemove = entryType switch
             {
-                var entriesToRemove = await Database.Incomes.Where(filter).Cast<IncomeModel>().ToListAsync();
-                Database.Incomes.RemoveRange(entriesToRemove);
-            }
-            else
-            {
-                var entriesToRemove = await Database.Expenses.Where(filter).Cast<ExpenseModel>().ToListAsync();
-                Database.Expenses.RemoveRange(entriesToRemove);
-            }
-
+                EntryType.Income => await Database.Incomes.Where(filter).Cast<IncomeModel>().ToListAsync(),
+                _ => await Database.Expenses.Where(filter).Cast<ExpenseModel>().ToListAsync()
+            };
+            Database.RemoveRange(entriesToRemove);
             await Database.SaveChangesAsync();
             return true;
         }
 
         public async Task<bool> DeleteListAsync(IEnumerable<int> idArray, int userId, EntryType entryType)
         {
-            if (entryType == EntryType.Income)
+            IEnumerable<IEntryModel> entriesToRemove = entryType switch
             {
-                var list = new List<IncomeModel>();
-                foreach (var id in idArray)
-                {
-                    var temp = await Database.Incomes.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId);
-                    list.Add(temp);
-                }
-                Database.Incomes.RemoveRange(list);
-            }
-            else
-            {
-                var list = new List<ExpenseModel>();
-                foreach (var id in idArray)
-                {
-                    var temp = await Database.Expenses.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId);
-                    list.Add(temp);
-                }
-                Database.Expenses.RemoveRange(list);
-            }
-
+                EntryType.Income => await Task.WhenAll(idArray.Select(selector: async id =>
+                    await Database.Incomes.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId))),
+                _ => await Task.WhenAll(idArray.Select(selector: async id =>
+                    await Database.Expenses.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId)))
+            };
+            Database.RemoveRange(entriesToRemove);
```

Comment:
Same here

Reply:
Thank you for asking about this, because this led me to some big improvements for this part :D Before this was generating threading issues, I think it's because of the two await statements, but I'm not completely sure why. 
Now the code looks like this:

            var filter = PredicateBuilder.BuildEntryFilter(idArray, userId);
            return await DeleteListAsync(filter, entryType);

You should refer to the new pull request about how this works, basically it creates a single predicate and does a single search, so not only does this work now ( tested it) also it should work faster, because of a query for every id in the list, now it just uses one query for all of them